python/src/iceberg/expressions.py

```diff
@@ -0,0 +1,106 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from enum import Enum
+
+from iceberg.exceptions import OperationError
```

Comment:
What is the convention you're using for spacing between `import` lines? Are they grouped?

Reply:
I deferred to isort for this but I believe all auto-sorters follow this recommendation from the [imports](https://www.python.org/dev/peps/pep-0008/#imports) section of the python style guide (PEP 8):
```
Imports are always put at the top of the file, just after any module comments and docstrings, and before module globals and constants.

Imports should be grouped in the following order:

- Standard library imports.
- Related third party imports.
- Local application/library specific imports.

You should put a blank line between each group of imports.
```
Also the next section in PEP 8 recommends absolute imports over relative imports for the third group (`from iceberg.exceptions import ...` as opposed to `from .exceptions import ...`). Unless anyone feels we should deviate from that, it might be worth adding to the contributing guidelines somewhere.